docs/features/software-catalog/external-integrations.md

```diff
@@ -1031,3 +1031,288 @@ backend.add(catalogModuleCustomDataParser);
+// This interface describes the type of data that will be passed to your burst function.
+interface Context {
+  apiClient: MyApiClient;
+}
+
+export class MyIncrementalEntityProvider
+  implements IncrementalEntityProvider<Cursor, Context>
+{
+  getProviderName() {
+    return `MyIncrementalEntityProvider`;
+  }
+}
+```
+
+`around` method is used for setup and tear-down. For example, if you need to create a client that will connect to the API, you would do that here.
+
+```ts
+export class MyIncrementalEntityProvider
+  implements IncrementalEntityProvider<Cursor, Context>
+{
+  getProviderName() {
+    return `MyIncrementalEntityProvider`;
+  }
+
+  async around(burst: (context: Context) => Promise<void>): Promise<void> {
+    const apiClient = new MyApiClient();
+
+    await burst({ apiClient });
+
+    // If you need to do any teardown, you can do it here.
+  }
+}
+```
+
+If you need to pass a token to your API, then you can create a constructor that will receive a token and use the token to setup the client.
+
+```ts
+export class MyIncrementalEntityProvider
+  implements IncrementalEntityProvider<Cursor, Context>
+{
+  token: string;
+
+  constructor(token: string) {
+    this.token = token;
+  }
+
+  getProviderName() {
+    return `MyIncrementalEntityProvider`;
+  }
+
+  async around(burst: (context: Context) => Promise<void>): Promise<void> {
+    const apiClient = new MyApiClient(this.token);
+
+    await burst({ apiClient });
+  }
+}
+```
+
+The last step is to implement the actual `next` method that will accept the cursor, call the API, process the result and return the result.
+
+```ts
+export class MyIncrementalEntityProvider implements IncrementalEntityProvider<Cursor, Context> {
+
+  token: string;
+
+  constructor(token: string) {
+    this.token = token;
+  }
+
+  getProviderName() {
+    return `MyIncrementalEntityProvider`;
+  }
+
+
+  async around(burst: (context: Context) => Promise<void>): Promise<void> {
+
+    const apiClient = new MyApiClient(this.token)
+
+    await burst({ apiClient })
+  }
+
+  async next(context: Context, cursor?: Cursor = { page: 1 }): Promise<EntityIteratorResult<Cursor>> {
+    const { apiClient } = context;
+
+    // call your API with the current cursor
+    const data = await apiClient.getServices(cursor);
+
+    // calculate the next page
+    const nextPage = page + 1;
+
+    // figure out if there are any more pages to fetch
+    const done = nextPage > data.totalPages;
+
+    // convert returned items into entities
+    const entities = data.items.map(item => ({
+      entity: {
+        apiVersion: 'backstage.io/v1beta1',
+        kind: 'Component',
+        metadata: {
+          name: item.name,
+          annotations: {
+            // You need to define these, otherwise they'll fail validation
+            [ANNOTATION_LOCATION]: this.getProviderName(),
+            [ANNOTATION_ORIGIN_LOCATION]: this.getProviderName(),
+          }
+        }
+        spec: {
+          type: 'service'
```

Comment:
```suggestion
          type: 'service',
          lifecycle: 'production', // Ideally your source has this information
          owner: 'unknown', // Ideally your source has this information
```

these are required fields, might as well make the example compile and ingest fully successfully